packages/xdl/src/Project.js

```diff
@@ -2097,13 +2096,14 @@ export async function optimizeAsync(projectRoot: string = './', options: Object
     const { size: prevSize } = fs.statSync(image);
 
     const newName = createNewFilename(image);
-    await optimizeImageAsync(image, newName, quality);
+    const optimizedImage = await optimizeImageAsync(image, quality);
 
-    const { size: newSize } = fs.statSync(image);
+    const { size: newSize } = fs.statSync(optimizedImage);
     const amountSaved = prevSize - newSize;
-    if (amountSaved < 0) {
-      // Delete the optimized version and revert changes
-      fs.renameSync(newName, image);
+    if (amountSaved > 0) {
+      await fs.move(image, newName);
```

Comment:
Promise.all

Reply:
These operations should not happen in parallel – we first move the old file `filename.orig.ext` and then move the optimized image from a temporary folder to its place.